game/songs.cc

```diff
@@ -509,39 +516,37 @@ void Songs::sortChange(int diff) {
 }
 
 void Songs::sortSpecificChange(int sortOrder, bool descending) {
-	if(sortOrder < 0) {
+	if(sortOrder < 0) 
 		m_order = 0;
-	} else if(sortOrder <= 6) {
+	else if(sortOrder < orders)
 		m_order = sortOrder;
-	} else {
+	else
 		m_order = 0;
-	}
+	
 	RestoreSel restore(*this);
 	config["songs/sort-order"].i() = m_order;
 	sort_internal(descending);
 }
 
 void Songs::sort_internal(bool descending) {
-	if(descending) {
-		switch (m_order) {
-		  case 0: std::stable_sort(m_filtered.begin(), m_filtered.end(), customComparator(&Song::randomIdx)); break;
-		  case 1: std::sort(m_filtered.rbegin(), m_filtered.rend(), customComparator(&Song::collateByTitle)); break;
-		  case 2: std::sort(m_filtered.rbegin(), m_filtered.rend(), customComparator(&Song::collateByArtist)); break;
-		  case 3: std::sort(m_filtered.rbegin(), m_filtered.rend(), customComparator(&Song::edition)); break;
-		  case 4: std::sort(m_filtered.rbegin(), m_filtered.rend(), customComparator(&Song::genre)); break;
-		  case 5: std::sort(m_filtered.rbegin(), m_filtered.rend(), customComparator(&Song::path)); break;
-		  case 6: std::sort(m_filtered.rbegin(), m_filtered.rend(), customComparator(&Song::language)); break;
-		  default: throw std::logic_error("Internal error: unknown sort order in Songs::sortChange");
-		}
-	} else {
+	if(m_order == 0)
+		std::stable_sort(m_filtered.begin(), m_filtered.end(), customComparator(&Song::randomIdx, true));
+	else {        
+		auto begin = m_filtered.begin();
+		auto end = m_filtered.end();
+
 		switch (m_order) {
-		  case 0: std::stable_sort(m_filtered.begin(), m_filtered.end(), customComparator(&Song::randomIdx)); break;
-		  case 1: std::sort(m_filtered.begin(), m_filtered.end(), customComparator(&Song::collateByTitle)); break;
-		  case 2: std::sort(m_filtered.begin(), m_filtered.end(), customComparator(&Song::collateByArtist)); break;
-		  case 3: std::sort(m_filtered.begin(), m_filtered.end(), customComparator(&Song::edition)); break;
-		  case 4: std::sort(m_filtered.begin(), m_filtered.end(), customComparator(&Song::genre)); break;
-		  case 5: std::sort(m_filtered.begin(), m_filtered.end(), customComparator(&Song::path)); break;
-		  case 6: std::sort(m_filtered.begin(), m_filtered.end(), customComparator(&Song::language)); break;
+		  case 1: std::sort(begin, end, customComparator(&Song::collateByTitle, !descending)); break;
+		  case 2: std::sort(begin, end, customComparator(&Song::collateByArtist, !descending)); break;
+		  case 3: std::sort(begin, end, customComparator(&Song::edition, !descending)); break;
+		  case 4: std::sort(begin, end, customComparator(&Song::genre, !descending)); break;
+		  case 5: std::sort(begin, end, customComparator(&Song::path, !descending)); break;
+		  case 6: std::sort(begin, end, customComparator(&Song::language, !descending)); break;
+		  case 7: std::sort(begin, end, [this, descending](SongPtr const& a, SongPtr const& b){
```

Comment:
capture descending by ref?

Reply:
Would work, but as descending is a bool a reference will have no benefit. Or am I wrong?

If reference it should de a const ref.